pandas/core/indexes/base.py

```diff
@@ -1278,39 +1278,46 @@ def _set_names(self, values, level=None):
 
     def set_names(self, names, level=None, inplace=False):
         """
-        Set new names on index. Defaults to returning new index.
+        Set Index or MultiIndex name.
+
+        Able to set new names partially and by level.
 
         Parameters
         ----------
-        names : str or sequence
-            name(s) to set
-        level : int, level name, or sequence of int/level names (default None)
-            If the index is a MultiIndex (hierarchical), level(s) to set (None
-            for all levels).  Otherwise level must be None
-        inplace : bool
-            if True, mutates in place
+        names : int, str, list-like, optional
+            Name(s) to set.
+        level : int, str, list-like, optional, default None
+            If the index is a MultiIndex(hierarchical), level(s) to set (None
+            for all levels). Otherwise level must be None.
+        inplace : boolean, default False
+            Modifies the object directly, instead of creating a new Index or
+            MultiIndex.
 
         Returns
         -------
-        new index (of same type and class...etc) [if inplace, returns None]
+        renamed : Index, MultiIndex or None
+            The same type as the caller or None if inplace is True.
+
+        See also
+        --------
+        Index.rename : Able to set new names without level
 
         Examples
         --------
         >>> pd.Index([1, 2, 3, 4]).set_names('foo')
         Int64Index([1, 2, 3, 4], dtype='int64', name='foo')
```

Comment:
I'd assign the index to a variable `idx`, display it, and then display it after setting the name. This example gives the impression that this is the right way to create an `Index` with a name.